shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/core/presentation/BirtdayDialog.kt

```diff
@@ -0,0 +1,40 @@
+package com.vml.tutorial.plantshop.core.presentation
+
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import com.vml.tutorial.plantshop.MR
+import com.vml.tutorial.plantshop.core.utils.exts.orZero
+import kotlinx.datetime.Clock
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun BirthdayDialog(onDateConfirm: (Long) -> Unit, onDismissRequest: () -> Unit) {
+    val eighteenYearsInMillis = 568024668000
+    val datePickerState =
```

Comment:
eighteenYearsInMillis could be private,

One more thing -> If you already set your birthday show it instead of initial date :) Right now if I save my birthday and I reopen the birthday picker it shows me the initial value (now - 18 years)